local-cli/runAndroid/runAndroid.js

```diff
@@ -200,6 +209,88 @@ function startServerInNewWindow() {
   }
 }
 
+
+function findActivityName(content, activityClass) {
+  const regex = new RegExp(`android:name="(.+?\.${activityClass})"`);
+  return content.match(regex)[1];
+}
+
+function findPreviousTerm(content, endPos) {
+  while (content[endPos] === ' ') {
+    --endPos;
+  }
+  const regex = new RegExp('\\w');
+  const word = [];
+  while (regex.exec(content[endPos])) {
+    word.push(content[endPos]);
+    --endPos;
+  }
+  return word.reverse().join('');
+}
+
+function findVariants(filePath, variantType, defaultVariants) {
+  const content = fs.readFileSync(filePath, 'utf8');
+  const regex = new RegExp(`${variantType}\\s+{`, 'ig');
+  const variants = defaultVariants || [];
+  const match = regex.exec(content);
+  if (!match) {
+    return variants;
+  }
+  const variantStartPos = regex.lastIndex;
+  let counter = 1;
+  let pos = variantStartPos + 1;
+  while (counter > 0) {
+    if (content[pos] === '{') {
+      counter += 1;
+      if (counter === 2) {
+        const previousTerm = findPreviousTerm(content, pos - 1);
+        if (variants.indexOf(previousTerm) === -1) {
+          variants.push(previousTerm);
+        }
+      }
+    } else if (content[pos] === '}') {
+      --counter;
+    }
+    ++pos;
+  }
+  return variants;
+}
+
+function splitVariant(gradleFilePath, variant) {
+  const buildTypes = findVariants(gradleFilePath, 'buildTypes', ['debug', 'release']);
+  const regexp = new RegExp(buildTypes.join('|'), 'gi');
+  const match = regexp.exec(variant);
+  if (match) {
+    return [variant.substring(match.index), variant.substring(0, match.index)];
+  }
+  return [variant, null];
+}
+
+function isSeparateBuildEnabled(gradleFilePath) {
+  const content = fs.readFileSync(gradleFilePath, 'utf8');
+  const separateBuild = content.match(/enableSeparateBuildPerCPUArchitecture\s+=\s+([\w]+)/)[1];
+  return separateBuild.toLowerCase() === 'true';
+}
+
+function getManifestFile(variant) {
+  const gradleFilePath = 'android/app/build.gradle';
+  const ret = splitVariant(gradleFilePath, variant);
+  const buildType = ret[0];
+  const flavor = ret[1];
+  const paths = ['android/app/build/intermediates/manifests/full'];
```

Comment:
Since we are already in the android directory: https://github.com/facebook/react-native/pull/8950/files#diff-a003f89d34db18b4567fdb55080a003eR81

I believe these two file paths should be `app/build.gradle` and `app/build/intermediates/manifests/full`